python/dgl/transform.py

```diff
@@ -404,6 +405,56 @@ def laplacian_lambda_max(g):
                                       return_eigenvectors=False)[0].real)
     return rst
 
+def coalesce_metapath(g, metapath):
+    """Transform a heterogeneous graph into either a homogeneous graph or a unidirectional
+    bipartite graph so that metapaths are coalesced into edges.
+
+    If the beginning node type ``s`` and ending node type ``t`` are the same, it will return
+    a homogeneous graph with node type ``s = t``.  Otherwise, a unidirectional bipartite graph
+    with source node type ``s`` and destination node type ``t`` is returned.
+
+    In both cases, two nodes ``u`` and ``v`` will be connected with an edge ``(u, v)`` if
+    there exists one path matching the metapath from ``u`` to ``v``.
+
+    Even if a node of type ``s`` does not have any outbound path matching the given metapath, it
+    will still show up in the returned graph.  Similarly, a node of type ``t`` will still
+    show up even if it does not have any inbound path matching the given metapath.
```

Comment:
The result graph keeps the node set of type ``s`` and ``t`` in the original graph even if they might have no neighbor.

Reply:
Much better...